### PR TITLE
Floating IP attach fix: fip by ID + instance by name and vice versa

### PR DESCRIPTION
**Pull request by david-crespo**
When I tried attaching a floating IP to an instance in the web console on dogfood, I got this error from the API.

It turns out we were taking the project as specified for determining the floating IP and jamming it into the instance selector, regardless of whether it needed it (selecting instance by name) or not (selecting instance by ID). I added a bit to the floating IP attach test and confirmed that it fails without the fix and passes with the fix.

Even with this fix in place, there is still one more loose end to tie up: if the floating IP is specified by ID, and therefore does not have the project, but the _instance_ is specified by name, this will fail because there is no project to include for instance specification purposes. The only way I can see of making this work without asking the user to somehow specify the project for the instance is to fetch the floating IP and pull the project off of that to include on the instance selector. This makes sense because you can only ever attach a floating IP to an instance in the same project.

Edit: did that in 6c144e82e04e1a64da9bd1a6ccc6026efb76f98e. Kinda sad to do an extra DB fetch, but I don't see another way to preserve the ability to do name or ID on the instance.

### CODE REVIEW

**Review by charliepark — COMMENTED** [review]
Looks good to me, but will defer to @FelixMcFelix for final 👍

**Review by FelixMcFelix — APPROVED** [review]
Good catch, and thanks for fixing this!

I think I'm going to take some time once this is merged to work on plumbing through a `FloatingIpLookup`/`authz::FloatingIp` to `instance_attach_external_ip` to make sure we're never resolving by name+project twice.

File: nexus/src/app/external_ip.rs
Comment (david-crespo) [reply nexus/src/app/external_ip.rs:171]: Oh yes, that is clearer. Thanks!

**Pull request merged**
